arch/arc/soc/snps_emsk/linker.ld

```diff
@@ -1,33 +1,40 @@
 /*
- * Copyright (c) 2016 Synopsys, Inc. All rights reserved.
+ * Copyright (c) 2018 Synopsys, Inc. All rights reserved.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 /**
- * @brief Linker script for the Synopsys EM Starterkit v2.2 EM11D platform.
+ * @brief Linker script for the Synopsys EM Starterkit platform.
  */
 
+#include <generated_dts_board.h>
+
 /*
  * DRAM base address and size
  *
  * DRAM includes the exception vector table at reset, which is at
  * the beginning of the region.
  */
+#if defined(CONFIG_SRAM_BASE_ADDRESS) && (CONFIG_SRAM_SIZE > 0)
```

Comment:
What's the purpose of these #ifdefs? there are no #else with alternatives or warning/error messages.
Also, the "TODO" comment below on SRAM doesn't seem to be a TODO, or any plans to change/update?

Reply:
it's used to avoid wrong definition in include\arc\arc\v2\linker.ld
If CONFIG_SRAM_BASE_ADDRESS is not defined.  without #if defined here,  no matter how, SRAM_START will be defined to CONFIG_SRAM_BASE_ADDRESS,  then region SRAM in MEMORY of linker.ld will be generated. This is a wrong definition.